lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1218,6 +1219,9 @@ static bool performCompileStepsPostSILGen(
   SILOptions &SILOpts = Invocation.getSILOptions();
   IRGenOptions &IRGenOpts = Invocation.getIRGenOptions();
 
+  BufferIndirectlyCausingDiagnosticRAII cpi(Context.Diags,
+                                            PSPs.MainInputFilenameForDebugInfo);
```

Comment:
Eek. Now it's no longer "for debug info". I don't think this is safe to use because the name for debug info may not be the same as the name of the buffer.

Reply:
Thanks--great catch! I switched it to use the MSF when it's a source file. Does that look right?

Reply:
Tx.